src/memory-analyzer/analyze_symbol.cpp

```diff
@@ -29,6 +31,66 @@ gdb_value_extractort::gdb_value_extractort(
 {
 }
 
+bool gdb_value_extractort::memory_scopet::contains(
+  const memory_addresst &point) const
+{
+  size_t begin_int = std::strtoul(begin.address_string.c_str(), NULL, 0);
+  size_t point_int = std::strtoul(point.address_string.c_str(), NULL, 0);
+  return point_int >= begin_int && (begin_int + byte_size) > point_int;
+}
+
+mp_integer gdb_value_extractort::memory_scopet::distance(
+  const memory_addresst &point,
+  mp_integer member_size) const
+{
+  CHECK_RETURN(contains(point));
+  size_t begin_int = std::strtoul(begin.address_string.c_str(), NULL, 0);
+  size_t point_int = std::strtoul(point.address_string.c_str(), NULL, 0);
+  return (point_int - begin_int) / member_size;
```

Comment:
Shouldn't you first do `CHECK_RETURN(point_int >= begin_int);`?

Reply:
That was implied by `contains`. But I refactored this section a bit to avoid duplicate computation of the address-to-string conversion. See: https://github.com/diffblue/cbmc/pull/4578/commits/3812c28519a207a0e5f247f4b7fe2cd4db19af2b.